blocks/shared-styles/_children/button/utils/getButtonStyle.js

```diff
@@ -0,0 +1,22 @@
+import { BUTTON_STYLES } from '../index';
+
+function getNavSpecificPrimaryButtonTheme(navColor, navBarBackground) {
+  if (navBarBackground === 'primary-color') {
+    return BUTTON_STYLES.PRIMARY_REVERSE;
+  }
+
+  return navColor === 'light' ? BUTTON_STYLES.PRIMARY : BUTTON_STYLES.DEFAULT;
+}
+
+function getNavSpecificSecondaryButtonTheme(navColor, navBarBackground) {
+  if (navBarBackground === 'primary-color' || navColor === 'dark') {
```

Comment:
What happens if the primary is not 'dark', but lighter color?  This kind of goes hand in hand with the other comment I put here.  If we have a Primary-color and a primary-color-reverse I would imagine those would be inverse for the background:color relationship.

Reply:
To follow up from slack - Primary button in the current implement is assumed to use the primary color but that will not always be the case. Primary Button and Primary Color are different - the primary button is the design it just so happens the base implementation uses the primary color - longer term the color used in primary button could be any color decided by a variable

Reply:
We currently have the same issue with the result list and search result list block "see more" button - it's an issue with the blocks assuming primary color is going to work with the predefined font color